packages/create-waku/src/cli.ts

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable */
+
+import fs from "node:fs";
+import path from "node:path";
+import prompts from "prompts";
+import { red, green, bold } from "kolorist";
+import fse from "fs-extra/esm";
+
+function isValidPackageName(projectName: string) {
+  return /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(
+    projectName,
+  );
+}
+
+function toValidPackageName(projectName: string) {
+  return projectName
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, "-")
+    .replace(/^[._]/, "")
+    .replace(/[^a-z0-9-~]+/g, "-");
+}
+
+// if the dir is empty or not exist
+function canSafelyOverwrite(dir: string) {
+  return !fs.existsSync(dir) || fs.readdirSync(dir).length === 0;
+}
+
+async function init() {
+  let targetDir = "";
+  const defaultProjectName = "waku-project";
+
+  const CHOICES = fs.readdirSync("template");
+  let result: {
+    packageName: string;
+    shouldOverwrite: string;
+    chooseProject: string;
+  };
+
+  try {
+    result = await prompts(
+      [
+        {
+          name: "projectName",
+          type: "text",
+          message: "Project Name",
+          initial: defaultProjectName,
+          onState: (state: any) =>
+            (targetDir = String(state.value).trim() || defaultProjectName),
+        },
+        {
+          name: "shouldOverwrite",
+          type: () => (canSafelyOverwrite(targetDir) ? null : "confirm"),
+          message: `${targetDir} is not empty. Remove existing files and continue?`,
+        },
+        {
+          name: "overwriteChecker",
+          type: (values: any) => {
+            if (values === false) {
+              throw new Error(red("✖") + " Operation cancelled");
+            }
+            return null;
+          },
+        },
+        {
+          name: "packageName",
+          type: () => (isValidPackageName(targetDir) ? null : "text"),
+          message: "Package name",
+          initial: () => toValidPackageName(targetDir),
+          validate: (dir: string) =>
+            isValidPackageName(dir) || "Invalid package.json name",
+        },
+        {
+          name: "chooseProject",
+          type: "select",
+          message: "Choose a starter template",
+          choices: [
+            { title: "basic-template", value: CHOICES[0] },
+            { title: "async-template", value: CHOICES[1] },
+            { title: "promise-template", value: CHOICES[2] },
+          ],
+        },
+      ],
+      {
+        onCancel: () => {
+          throw new Error(red("✖") + " Operation cancelled");
+        },
+      },
+    );
+  } catch (cancelled) {
+    if (cancelled instanceof Error) {
+      console.log(cancelled.message);
+    }
+    process.exit(1);
+  }
+
+  const { packageName, shouldOverwrite, chooseProject } = result;
+
+  const root = path.resolve(targetDir);
+
+  if (shouldOverwrite) {
+    fse.emptyDirSync(root);
+  } else if (!fs.existsSync(root)) {
+    fs.mkdirSync(root);
```

Comment:
Curious if we should have the recursive option:
https://nodejs.org/dist/latest-v18.x/docs/api/fs.html#fsmkdirsyncpath-options

Reply:
Yes, good for creating nested dir.